org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleBuildSupport.java

```diff
@@ -106,7 +107,17 @@ public void update(IProject project, boolean force, IProgressMonitor monitor) th
 					|| (settingsFile.exists() && JavaLanguageServerPlugin.getDigestStore().updateDigest(settingsFile.toPath()))
 					|| (buildKtsFile.exists() && JavaLanguageServerPlugin.getDigestStore().updateDigest(buildKtsFile.toPath()))
 					|| (settingsKtsFile.exists() && JavaLanguageServerPlugin.getDigestStore().updateDigest(settingsKtsFile.toPath()));
-			if (isRoot || shouldUpdate) {
+			// https://github.com/redhat-developer/vscode-java/issues/3893
+			shouldUpdate = isRoot || shouldUpdate;
```

Comment:
This can be `shouldUpdate |= isRoot` right ?

Reply:
Fixed.

Reply:
Wait, you made it `shouldUpdate |= shouldUpdate`. Should be what I mentioned above I think.

Reply:
Sorry. Fixed.